mutiny-core/src/nostr/nwc.rs

```diff
@@ -27,11 +29,57 @@ pub struct SingleUseSpendingConditions {
     pub amount_sats: u64,
 }
 
+#[derive(Debug, Clone, PartialEq, Eq, Hash, Serialize, Deserialize)]
+pub struct TrackedPayment {
+    /// Time in seconds since epoch
+    time: u64,
+    /// Amount in sats
+    amt: u64,
+    /// Payment hash
+    hash: String,
+}
+
+#[derive(Debug, Clone, PartialEq, Eq, Hash, Serialize, Deserialize)]
+pub struct BudgetedSpendingConditions {
+    /// Amount in sats for the allotted budget period
+    pub budget: u64,
+    /// Max amount in sats for a single payment
+    pub single_max: Option<u64>,
+    /// Payment history
+    pub payments: Vec<TrackedPayment>,
+    /// Time period in seconds
+    pub period: u64,
```

Comment:
Is the idea that it is over the last X amount of seconds? So instead of a "monthly" budget, it's more of a over the last 30 days kind of budget?

Reply:
Yeah exactly, for testing I was using a 60 second budget but I am expecting for us to expose like daily options

Reply:
I think it would be good to do something that is interval based instead of past X time. If it were specific then you could do accurate monthly invoicing on the server side. Past 30 days doesn't really map to any real world logic. We would have this problem with Mutiny+ since we do 3 days early. We should remove that and be more precise and match that logic here. Would allow Mutiny+ to be auto approval with budget if a user wanted. Or other services.